geomesa-kafka/geomesa-kafka-geoserver-plugin/src/main/scala/org/locationtech/geomesa/kafka/plugin/ReplayKafkaDataStoreProcess.scala

```diff
@@ -0,0 +1,143 @@
+              @DescribeParameter(name = "endTime",   description = "End Time of the replay window")
+              endTime: Integer,   //Todo: change to something that can be used for Joda constructor
+              @DescribeParameter(name = "readBehind", description = "The amount of time to pre-read")
+              readBehind: java.lang.Long
+              ): String = {
+
+    val workspaceInfo: WorkspaceInfo = getWorkspace(workspace)
+
+    val catalogBuilder = new CatalogBuilder(catalog)
+    catalogBuilder.setWorkspace(workspaceInfo)
+
+    //TODO: Make sure we create the new store, parse replay config things!
+    val newStore = catalog.getFactory.createDataStore()
+    newStore.setWorkspace(workspaceInfo)
+    newStore.setName(store) // possibly the store parameter given above, or can be removed
+    newStore.setDescription("Volatile Kafka Replay Layer")
+    newStore.setType("Kafka Replay Data Store") // TODO: this is naive...
+    newStore.setEnabled(true)
+    catalog.add(newStore) // TODO: should I be using catalog.add or catalogBuilder.attach?
+
+    // grab out the Store we just added, this may be unnecessary
+    val storeInfo: DataStoreInfo = Option(catalog.getDataStoreByName(workspaceInfo.getName, store)).getOrElse {
+      throw new ProcessException(s"Unable to find store $store in workspace $workspace")
+    }
+    //create volatile SFT, todo: will need to use replay config here
+    val volatileSFT = createVolatileSFT(features, storeInfo, workspaceInfo.getName)
+
+    // set the catalogBuilder to our store
+    catalogBuilder.setStore(storeInfo)
+
+    // add well known volatile keyword
+    val volatileTypeInfo = catalogBuilder.buildFeatureType(volatileSFT.getName)
+    val volatileKW: Keyword = new Keyword("kafka.geomesa.volatile") //Todo: make this accessible to other things
+    volatileTypeInfo.getKeywords.add(volatileKW)
+
+    // do some setup
+    catalogBuilder.setupBounds(volatileTypeInfo)
+
+    // build the layer and add it to geoserver
+    val volatileLayerInfo = catalogBuilder.buildLayer(volatileTypeInfo)
+    catalog.add(volatileTypeInfo)
+    catalog.add(volatileLayerInfo)
+
+    s"created workspace: ${workspaceInfo.getName} and layer: ${volatileLayerInfo.getName}"
+  }
+
+  private def createVolatileSFT(features: SimpleFeatureCollection,
+                                storeInfo: DataStoreInfo,
+                                workspace: String): SimpleFeatureType = {
+    // TODO: wire in bits for replay config
+    // get source layer SFT
+    val extantSFT: SimpleFeatureType = features.getSchema
+    val extantSftName = extantSFT.getTypeName
+    // create new sft
+    val targetSftName = extantSftName+s"_volatile_Something" // todo: figure out new sft name
+    val sftBuilder = new SimpleFeatureTypeBuilder()
+    sftBuilder.init(extantSFT)
+    sftBuilder.setName(targetSftName)
+    val destinationSFT = sftBuilder.buildFeatureType()
```

Comment:
The destinationSFT can be created using the utility method that Matt is working on.